api/v1alpha1/sentinel_types.go

```diff
@@ -0,0 +1,191 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	"github.com/3scale/saas-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	// SentinelPort is the port where sentinel process listens
+	SentinelPort uint32 = 26379
+)
+
+// bitnami/redis-sentinel:4.0.11-debian-9-r110
+var (
+	sentinelDefaultReplicas int32            = 3
+	sentinelDefaultImage    defaultImageSpec = defaultImageSpec{
+		Name:       pointer.StringPtr("bitnami/redis-sentinel"),
+		Tag:        pointer.StringPtr("4.0.11-debian-9-r110"),
+		PullPolicy: (*corev1.PullPolicy)(pointer.StringPtr(string(corev1.PullIfNotPresent))),
+	}
+	sentinelDefaultResources defaultResourceRequirementsSpec = defaultResourceRequirementsSpec{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("75m"),
+			corev1.ResourceMemory: resource.MustParse("64Mi"),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("150m"),
+			corev1.ResourceMemory: resource.MustParse("128Mi"),
+		},
+	}
+	sentinelDefaultProbe defaultProbeSpec = defaultProbeSpec{
+		InitialDelaySeconds: pointer.Int32Ptr(25),
+		TimeoutSeconds:      pointer.Int32Ptr(1),
+		PeriodSeconds:       pointer.Int32Ptr(10),
+		SuccessThreshold:    pointer.Int32Ptr(1),
+		FailureThreshold:    pointer.Int32Ptr(3),
+	}
+	sentinelDefaultPDB defaultPodDisruptionBudgetSpec = defaultPodDisruptionBudgetSpec{
+		MaxUnavailable: util.IntStrPtr(intstr.FromInt(1)),
+	}
+
+	sentinelDefaultGrafanaDashboard defaultGrafanaDashboardSpec = defaultGrafanaDashboardSpec{
+		SelectorKey:   pointer.StringPtr("monitoring-key"),
+		SelectorValue: pointer.StringPtr("middleware"),
+	}
+	sentinelDefaultStorageSize            string        = "10Mi"
+	sentinelDefaultMetricsRefreshInterval time.Duration = 30 * time.Second
+)
+
+// SentinelConfig defines configuration options for the component
+type SentinelConfig struct {
+	// Monitored shards indicates the redis servers that form
+	// part of each shard monitored by sentinel
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	MonitoredShards map[string][]string `json:"monitoredShards"`
+	// StorageClass is the storage class to be used for
+	// the persistent sentinel config file where the shards
+	// state is stored
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
```

Comment:
when being optional, notmally we add the the `omitempty`, isn't it?

In addition, in that case there is no default value, so it will take default cluster storage class, isn't it?

Reply:
Yep that is missing. It doesn't affect the functionality (the field is optional anyway because it's a pointer) but the json tag being missing could lead to weird serialization behaviors. Fixed with 2531271

Reply:
nice!

Regarding the second question, in that case there is no default value or storageClass, so it will take default cluster storage class, isn't it?

Reply:
yes, if the value for the storageclass in the volume templace is `nil` the cluster will assign the default storage class.